components/cluster-coordinator/org.wso2.carbon.cluster.coordinator/org.wso2.carbon.cluster.coordinator.zookeeper/src/main/java/org/wso2/carbon/cluster/coordinator/zookeeper/internal/ZookeeperCoordinationServiceComponent.java

```diff
@@ -0,0 +1,122 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.cluster.coordinator.zookeeper.internal;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.wso2.carbon.cluster.coordinator.commons.CoordinationStrategy;
+import org.wso2.carbon.cluster.coordinator.commons.configs.CoordinationPropertyNames;
+import org.wso2.carbon.cluster.coordinator.commons.exception.ClusterCoordinationException;
+import org.wso2.carbon.cluster.coordinator.zookeeper.ZookeeperCoordinationStrategy;
+import org.wso2.carbon.cluster.coordinator.zookeeper.exception.ZookeeperCoordinationConfigurationException;
+import org.wso2.carbon.kernel.configprovider.CarbonConfigurationException;
+import org.wso2.carbon.kernel.configprovider.ConfigProvider;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Zookeeper cluster coordinator data service.
+ */
+@Component(
+        name = "org.wso2.carbon.cluster.coordinator.zookeeper.internal.ZookeeperCoordinationServiceComponent",
+        immediate = true
+)
+public class ZookeeperCoordinationServiceComponent {
+
+    private static final Log log = LogFactory.getLog(ZookeeperCoordinationServiceComponent.class);
+
+    /**
+     * Config provider to read the configuration file
+     */
+    private Map clusterConfiguration;
+
+    /**
+     * This is the activation method of ZookeeperCoordinationServiceComponent. This will be called when its references
+     * are satisfied.
+     *
+     * @param bundleContext the bundle context instance of this bundle.
+     * @throws Exception this will be thrown if an issue occurs while executing the activate method
+     */
+    @Activate
+    protected void start(BundleContext bundleContext) {
+
+        log.info("Getting the Coordination Strategy");
+
+        Map<String, Object> clusterConfiguration = null;
+        try {
+            clusterConfiguration = ZookeeperCoordinationServiceHolder.getConfigProvider().
+                    getConfigurationMap(CoordinationPropertyNames.CLUSTER_CONFIG_NS);
+            ZookeeperCoordinationServiceHolder.setClusterConfiguration(clusterConfiguration);
+        } catch (CarbonConfigurationException e) {
+            throw new ClusterCoordinationException("Configurations for cluster coordination is not " +
+                    "available in deployment.yaml");
```

Comment:
Better to throw ClusterCoordinationException with 'e' of the 'CarbonConfigurationException'

Reply:
Fixed in a06a773